cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -109,6 +110,11 @@ func ResyncPeriod(s *options.CMServer) func() time.Duration {
 
 // Run runs the CMServer.  This should never exit.
 func Run(s *options.CMServer) error {
+	// To help debugging, immediately log version and command line flags.
+	glog.Infof("%+v", version.Get())
```

Comment:
I agree.  Sort of a pre-`Run()` that everything should call at startup.  Where do you think is the best place to put such a function?